orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -279,22 +282,32 @@ def _get_session_config(self):
         if self._device.type == 'cuda':
             # Configure the InferenceSessions to use the specific GPU on which the model is placed.
             providers = (["ROCMExecutionProvider"] if self.is_rocm_pytorch else [
-                         "CUDAExecutionProvider"])
+                        "CUDAExecutionProvider"])
             providers.append("CPUExecutionProvider")
             provider_option_map = {"device_id": str(self._device.index)}
             if not self.is_rocm_pytorch:
-                # Set Conv algo search mode to HEURISTIC, which is same as PyTorch's default setting.
+            # Set Conv algo search mode to HEURISTIC, which is same as PyTorch's default setting.
                 provider_option_map["cudnn_conv_algo_search"] = "HEURISTIC"
                 provider_option_map["cudnn_conv_use_max_workspace"] = "1"
                 provider_option_map["cudnn_conv1d_pad_to_nc1d"] = "1"
             if self._use_external_gpu_allocator:
                 provider_option_map["gpu_external_alloc"] = str(self._torch_alloc)
                 provider_option_map["gpu_external_free"] = str(self._torch_free)
                 provider_option_map["gpu_external_empty_cache"] = str(self._torch_empty_cache)
-            provider_options = [provider_option_map, {}]
+                provider_options = [provider_option_map, {}]
         elif self._device.type == 'cpu':
-            providers = ["CPUExecutionProvider"]
-            provider_options = [{}]
+            if self._provider_configs.provider == "openvino":
```

Comment:
i feel we can make it more general. the real thing we need here is to have a mapping from pytorch device to available OR EPs. a single pytorch device may map to multiple EPs, with priority, i.e. CPU device map to OpenVINO EP and CPU EP, cuda device map to TensorRT EP and cuda EP.  So instead of have a configuration on ORTModule for EPs, i prefer to have an option let user configure the available EP priorities, then ORTModule can figure out the correct mapping based on that. something like:

import torch
import onnxruntime

onnxruntime.set_ep_priority('OpenVINOEP', 'CPUEP')
...
model = ORTModule(model)
...

with the priority configuration, we can have a generic algo here to generate the correct mapping, without change ortmodule api.